### PR TITLE
Feature/share function

### PR DESCRIPTION
**Pull request by nirali25parekh**
Changes in this pull request:
- share urls for social media linked
1. email (with subject and body)
2. facebook (with url and caption text)
3. twitter (with tweet text)
4. linkedin (with text)  

Plant-for-the-Planet - Google Chrome 2020-09-09 18-07-54

@harshvitra

### CODE REVIEW

**Comment by sagararyal** [comment]
Screen Shot 2020-09-10 at 5 42 09 PM


Screen Shot 2020-09-10 at 5 42 23 PM

Screen Shot 2020-09-10 at 5 42 49 PM
  
Screen Shot 2020-09-10 at 5 43 10 PM

**Comment by sagararyal** [comment]
@nirali25parekh please use share buttons like this.
@sarveshpro can help with the font awesome variant of the icons.
Screen Shot 2020-09-10 at 5 45 03 PM

reference is available at the bottom of this page.
https://xd.adobe.com/view/8f1c5110-4d7d-445d-8283-8eb1674ce2e4-da4f/screen/12e48f82-7be4-4991-906f-7f5fb8b5c7ab

Also please include a download button instead of email: https://fontawesome.com/icons/download?style=regular

use font awesome regular for display and move to font awesome solid icon variant on hover.

**Comment by felixfinkbeiner** [comment]
Screenshot 2020-09-10 at 15 17 38

Let's have this kind of thing where the birds represent (in this order):

- Download
- facebook
- instagram
- twitter
- email

**Comment by nirali25parekh** [comment]
<br/>
The color changes on hover according to their respective colors. This is how it currently looks

<br/>

Plant-for-the-Planet - Google Chrome 2020-09-11 03-09-23

<br/>
<br/>

The download button outlined was in the premium package in FontAwesome version, so used solid for both.

<br/>



<br/>

**Comment by nirali25parekh** [comment]
- the share buttons frontend changes done
- jpeg download functionality is done

phone_download

**Comment by sagararyal** [comment]
Screen Shot 2020-09-13 at 9 54 48 PM

Screen Shot 2020-09-13 at 9 53 32 PM

**Comment by sagararyal** [comment]
Download on Safari 13.1.2 

Bug:
component-2

Screen Shot 2020-09-13 at 9 56 46 PM
Screen Shot 2020-09-13 at 9 56 41 PM
Screen Shot 2020-09-13 at 9 56 38 PM
Screen Shot 2020-09-13 at 9 56 34 PM

File: src/features/public/Donations/components/ThankYou.tsx [hunk 1]
```diff
@@ -89,26 +72,26 @@ function ThankYou({
...
-      <div className={styles.thankyouImageContainer}>
+      <div className={styles.thankyouImageContainer} ref={imageRef}>
         <div className={styles.thankyouImage}>
-          {/* <div className={styles.pfpLogo}>
-                        <PlanetLogo />
-                    </div> */}
+
           <div className={styles.donationCount}>
             My {treeCount} trees are being planted in {project.location}
             <div className={styles.donationTenant}>
```
Comment (sagararyal) [inline src/features/public/Donations/components/ThankYou.tsx:97]: plant trees at {{tenantURL}} there is a variable from ENV

File: src/features/public/Donations/components/ShareOptions.jsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+const titleToShare = 'Planting trees against the climate crisis!';
+const urlToShare =
+  process.env.TENANT === 'salesforce'
+    ? 'https://www.salesforce.com/'
+    : 'https://www.trilliontreecampaign.org/';
+const linkToShare =
+  process.env.TENANT === 'salesforce'
+    ? 'salesforce.com/trees'
+    : 'trilliontreecampaign.org';
```
Comment (sagararyal) [inline src/features/public/Donations/components/ShareOptions.jsx:25]: use {tenantUrl}

File: src/features/public/Donations/components/ShareOptions.jsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      try {
+        const response = await navigator.share({
+          title: titleToShare,
+          text: textToShare,
+        });
+      } catch (error) {}
+    } else {
+      if (isMobileBrowser()) {
+        // if in phone, copy to clipboard
+        navigator.clipboard.writeText('Dummy text copied to clipboard!');
```
Comment (sagararyal) [inline src/features/public/Donations/components/ShareOptions.jsx:43]: This needs to be complete.

File: src/features/public/Donations/components/ShareOptions.jsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        className={styles.shareIcon}
+        onClick={() => {
+          if (props.toPrintRef) {
+            rcei.exportComponentAsJPEG(props.toPrintRef);
+          }
+        }}
+        onMouseOver={() => setCurrentHover(1)}
+      >
+        {currentHover === 1 ? (
+          <DownloadSolid color="rgba(0,0,0,0.9)" />
```
Comment (sagararyal) [inline src/features/public/Donations/components/ShareOptions.jsx:64]: change colors to planet grey

**Review by sagararyal — CHANGES_REQUESTED** [review]

**Comment by nirali25parekh** [comment]
Changes:
- image download bug fixed on safari
- icons change > color, size, round border, layout
- redundant components removed
- env variables used tentantUrl

shareIcons

**Comment by Plant-for-the-Planet** [comment]
@nirali25parekh 
I still don't see the circles but that is perfectly fine.

- [ ] Need to increase the size/resolution of the downloaded image.

Screen Shot 2020-09-14 at 11 31 34 PM

**Comment by nirali25parekh** [comment]
changes:
- safari share links fixed
- file name changed
- image quality (I scaled the image from 1.0 to 5.0, but the downloaded image still seems blur. I'm looking into other ways it can be improved)

**Comment by felixfinkbeiner** [comment]
1. Please remove the horizontal line that separates the text from the share card

2. When clicking the twitter icon or the email icon on my android it opens the share card instead of twitter even though I have the twitter app installed (same for insta and fb but I don't have those apps)

3. Is there a legal reason we need "pay with Credit Card"? Or what was our motivation for including it. I don't think it's necessary. People know what they paid with and it just makes this text longer (and is grammatically incorrect)

The download works really well. Good job on managing to rename the downloaded file!!!

**Comment by felixfinkbeiner** [comment]
Oh and a couple more things:

1. In the download the logo and the trillion trees is super low res while the adjusted text looks great. But I think you know that

2. The plant trees at www.trillion does not need "www." And it should be right at the bottom of the card. Ideally, in the card preview it would be a bit further up (as it is now) to make space for the share buttons, but once you actually download it, the text would be further down. Is that possible?

3. In the share page, there is too much space between the bottom of the buttons and the bottom of the pop-up

**Comment by harshvitra** [comment]
> 3\. Is there a legal reason we need "pay with Credit Card"? Or what was our motivation for including it. I don't think it's necessary. People know what they paid with and it just makes this text longer (and is grammatically incorrect)

For Apple/Google pay we need it 
@sagararyal and I had a conversation on this long back, please confirm

**Comment by harshvitra** [comment]
**To Do -** 

- [ ] Please remove the horizontal line that separates the text from the share card
 
- [ ] When clicking the twitter icon or the email icon on my android it opens the share card instead of twitter even though I have the twitter app installed (same for insta and fb but I don't have those apps)

- [ ] Is there a legal reason we need "pay with Credit Card"? Or what was our motivation for including it. I don't think it's necessary. People know what they paid with and it just makes this text longer (and is grammatically incorrect) @sagararyal please comment

- [ ] In the download the logo and the trillion trees is super low res while the adjusted text looks great. But I think you know that
 
- [ ] The plant trees at www.trillion does not need "www." And it should be right at the bottom of the card. Ideally, in the card preview it would be a bit further up (as it is now) to make space for the share buttons, but once you actually download it, the text would be further down. Is that possible?
 
- [ ] In the share page, there is too much space between the bottom of the buttons and the bottom of the pop-up

**Comment by Plant-for-the-Planet** [comment]
Moving remaining tasks to a new issue and merging this to develop.

**Pull request merged**